core/src/main/java/eu/fasten/core/dependents/data/Dependency.java

```diff
@@ -0,0 +1,483 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.fasten.core.dependents.data;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import eu.fasten.core.data.Constants;
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
+ * A dependency declaration. Denotes a Revision's will to use the functionality of the
+ * {@class Product} that matches the dependency's qualifiers.
+ */
+public class Dependency extends Product {
+    public static final Dependency empty = new Dependency("", "");
+
+    public final List<VersionConstraint> versionConstraints;
+    public final List<Exclusion> exclusions;
+    public final String scope;
+    public final boolean optional;
+    public final String type;
+    public final String classifier;
+
+    /**
+     * Valid dependency scopes. Defined by maven.
+     * Learn more: http://maven.apache.org/pom.html
+     */
+    public static final String[] SCOPES = {
+            "compile",
+            "provided",
+            "runtime",
+            "test",
+            "system",
+            "import"
+    };
+
+    /**
+     * Constructor for Dependency object.
```

Comment:
Again, this is for maven. The method parameters may not be relevant for Python & C.

Reply:
You are right,  the only Dependency information currently retrieved from the C & Python Database is the package Name and the Version Constraints, so I will remove the unnecessary Dependency & Dependnecy Edge attributes.